homeassistant/components/sensor/wunderground.py

```diff
@@ -666,8 +679,12 @@ def __init__(self, hass: HomeAssistantType, rest, condition):
         self._entity_picture = None
         self._unit_of_measurement = self._cfg_expand("unit_of_measurement")
         self.rest.request_feature(SENSOR_TYPES[condition].feature)
+        current_ids = set(hass.states.async_entity_ids(sensor.DOMAIN))
+        current_ids |= hass.data[ADDED_ENTITY_IDS_KEY]
         self.entity_id = async_generate_entity_id(
```

Comment:
The real problem here in this component is that this entity is taking care of it's own entity ids. That is not the preferred approach. And guess what, because of this hack, we now need to add another hack to support entity namespace? Seems like a red flag to me…

Reply:
It is a hack - this whole platform is very hackish :/

I just don't see any other way we can do custom names + constant "stable" entity ids at the same time. I mean we could take a look at disabling custom names to clean up the platform. Or we could even at some point take a look at transitioning this to the weather component.

Reply:
I think that the real solution would be to do nothing with entity ids and just give it a unique id. Then people can rename it themselves.

For now let's keep it in, I guess? I don't really care that strongly.